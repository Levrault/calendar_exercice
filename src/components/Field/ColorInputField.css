.container {
  display: flex;
  position: relative;
  text-align: center;
}

.label {
  margin-top: $space-xs;
  pointer-events: none;
  position: absolute;
  transition: all .5s;
  width: 100%;
}

.input {
  background: transparent;
  border: 0;
  border-bottom: 2px solid $palette-accent;
  margin-bottom: $space-s;
  padding: $space-s;
  transition: all .5s;
  width: $space-xxl;

  &:focus {
    outline: 0;
  }

  &:focus + .label,
  &:valid + .label {
    color: $palette-secondary-text;
    transform: translateY(-resolve($space-m));
  }
}
