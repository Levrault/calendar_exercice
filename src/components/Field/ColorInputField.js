import React from 'react';
import PropTypes from 'prop-types';
import colors from '../Palette/ColorPalette';
import styles from './ColorInputField.css';

/**
 * @param {string} label
 */
const ColorInputField = ({ label, code, ...rest }) => {
  return (
    <div className={styles.container}>
      <input {...rest} className={styles.input} style={{ borderBottom: `2px solid ${code}` }} required />
      <label className={styles.label} htmlFor={name}>{label}</label>
    </div>
  );
};

ColorInputField.propTypes = {
  label: PropTypes.string.isRequired,
  code: PropTypes.string
};

ColorInputField.defaultProps = {
  code: colors[colors.length - 1]
};

export default ColorInputField;

