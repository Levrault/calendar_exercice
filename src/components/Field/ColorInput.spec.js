import React from 'react';
import renderer from 'react-test-renderer';
import ColorInputField from './ColorInputField';

describe('ColorInput component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <ColorInputField
        name="event"
        type="text"
        label="Rappel"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a different border color', () => {
    const component = renderer.create(
      <ColorInputField
        name="event"
        type="text"
        code="#FFFFFF"
        label="Rappel"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
