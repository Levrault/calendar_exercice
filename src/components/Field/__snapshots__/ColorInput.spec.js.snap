// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInput component should render with a different border color 1`] = `
<div>
  <input
    name="event"
    style={
      Object {
        "borderBottom": "2px solid #FFFFFF",
      }
    }
    type="text"
  />
  <label
    htmlFor="nodejs"
  >
    Rappel
  </label>
</div>
`;

exports[`ColorInput component should render without problem 1`] = `
<div>
  <input
    name="event"
    style={
      Object {
        "borderBottom": "2px solid undefined",
      }
    }
    type="text"
  />
  <label
    htmlFor="nodejs"
  >
    Rappel
  </label>
</div>
`;
